syntax/blocks.js

```diff
@@ -112,7 +107,7 @@ var allBlocks = scratchCommands.map(function(def) {
     spec: def.spec, // Used for Scratch 2 translations
     parts: def.spec.split(splitPat).filter(x => !!x),
     selector: def.selector || "sb3:" + def.id, // Used for JSON marshalling
-    inputs: def.inputs,
+    inputs: Array.isArray(def.inputs) ? def.inputs : [],
```

Comment:
Why is this needed, out of interest?

I think the following might be a little more correct, and would make it easier to debug if someone accidentally sets `inputs` to something that is _not_ an array:
```suggestion
    inputs: def.inputs == null ? [] : def.inputs,
```

Reply:
Yep, we can do this